datalad/core/distributed/push.py

```diff
@@ -772,13 +782,16 @@ def _push_data(ds, target, content, force, jobs, res_kwargs,
     if jobs:
         cmd.extend(['--jobs', str(jobs)])
 
-    if force not in ('pushall', 'datatransfer') and ds_repo.config.obtain(
-            'datalad.push.copy-auto-if-wanted'):
-        if ds_repo.get_preferred_content('wanted', target):
-            lgr.debug("Invoking copy --auto")
-            cmd.append('--auto')
+    # Since we got here - we already have some  data != "nothing"
+    if (data == 'auto') or \
+        (
+            (data == 'auto-if-wanted') and
+            ds_repo.get_preferred_content('wanted', target)
+        ):
+        lgr.debug("Invoking copy --auto")
+        cmd.append('--auto')
```

Comment:
Okay, so ignoring the obvious addition of the `--data` flag, the core change is here.  We switch the decision to pass `--auto` from

```
you didn't explicitly ask to transfer data and you set this config
```

to


```
you didn't explicitly ask to transfer data and you've configured wanted for the remote
```

As far as I can tell, that'd achieve both the behavior that @yarikoptic is after (respect "wanted") and @mih is after (just transfer the data).  It comes with an assumption that there aren't callers that set wanted yet would still prefer `push` to not use `--auto`; sounds pretty safe.

It swaps a `config.obtain` call for a `.get_preferred_content` call per dataset, which seems acceptable given any data transfer should swamp that expense.